### PR TITLE
update ci config to support build and delivery from overlay

### PR DESCRIPTION
**Pull request by harshad16**
update ci config to support build and delivery from overlay
Signed-off-by: Harshad Reddy Nalla <hnalla@redhat.com>

### CODE REVIEW

File: .aicoe-ci.yaml [hunk 1]
```diff
@@ -1,15 +1,37 @@
...
+
+  - name: inference
+    build:
+      base-image: quay.io/thoth-station/s2i-thoth-ubi8-py38:v0.20.1
+      build-stratergy: Source
+      registry: quay.io
+      registry-org: thoth-station
+      registry-project: elyra-aidevsecops-tutorial
+      registry-secret: thoth-station-thoth-pusher-secret
+    deploy:
```
Comment (goern) [inline .aicoe-ci.yaml:33]: this is meant as a deploy per overlay?

Comment (pacospace) [reply .aicoe-ci.yaml:33]: I think it deployed just the overlay you want to be deployed, in this case only the inference one. From Elyra point of view you don't need deployment of the other steps, as they are pulled from registry during Kubeflow Pipeline Run

File: .aicoe-ci.yaml [hunk 2]
```diff
@@ -1,15 +1,37 @@
...
+  - name: download-dataset
+    build:
+      base-image: quay.io/thoth-station/s2i-thoth-ubi8-py38:v0.20.1
+      build-stratergy: Source
+      registry: quay.io
+      registry-org: thoth-station
+      registry-project: elyra-aidevsecops-dataset
+      registry-secret: thoth-station-thoth-pusher-secret
+
+  - name: test-model
```
Comment (pacospace) [inline .aicoe-ci.yaml:16]: added missing overlay for `training`, we don't need test_model image for now. and I need to be sure also the overlays have correct naming withou `_` right?

Comment (harshad16) [reply .aicoe-ci.yaml]: yes thats correct, for now  i have added that change to this PR.

File: .aicoe-ci.yaml (hunk 1 above)
Comment (harshad16) [reply .aicoe-ci.yaml:33]: yes, we can have deploy the overlay to different places. 
So aicoe-ci supports the delivery per overlay.

**Pull request merged**